File: app/src/main/java/com/openvehicles/OVMS/entities/StoredCommand.java
package com.openvehicles.OVMS.entities;

import androidx.annotation.Nullable;

public class StoredCommand
	implements Comparable<StoredCommand> {

	public long mKey;
	public String mTitle;
	public String mCommand;

	public StoredCommand(long pKey, String pTitle, String pCommand) {
		this.mKey = pKey;
		this.mTitle = pTitle;
		this.mCommand = pCommand;
	}
	public StoredCommand(String pTitle, String pCommand) {
		this(0, pTitle, pCommand);
	}

	@Override
	public int compareTo(StoredCommand o) {
		return mTitle.compareToIgnoreCase(o.mTitle);
	}

	@Override
	public boolean equals(@Nullable Object obj) {
		if (obj instanceof StoredCommand) {
			return (((StoredCommand)obj).mKey == this.mKey);
		} else {
			return false;
		}
	}
}

File: app/src/main/java/com/openvehicles/OVMS/entities/StoredCommand.kt
package com.openvehicles.OVMS.entities

import java.io.Serializable

class StoredCommand(

    @JvmField
    var key: Long,
    @JvmField
    var title: String,
    @JvmField
    var command: String

) : Comparable<StoredCommand>, Serializable {

    constructor(title: String, command: String) : this(0, title, command)

    override fun compareTo(other: StoredCommand): Int {
        return title.compareTo(other.title, ignoreCase = true)
    }

    override fun equals(other: Any?): Boolean {
        return if (other is StoredCommand) {
            other.key == key
        } else {
            false
        }
    }
}
